### PR TITLE
ehco: update to build with go@1.20

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/e/ehco.rb [hunk 1]
```diff
@@ -22,22 +23,21 @@ class Ehco < Formula
...
 
   uses_from_macos "netcat" => :test
 
-  # Backport quic-go update to support Go 1.19
-  # Remove in the next release
+  # go@1.20 build patch, remove in next release
   patch do
-    url "https://github.com/Ehco1996/ehco/commit/2f739a4279f9defaeb8beac9e97e82851e0dd995.patch?full_index=1"
-    sha256 "43235566b344e0d125d79719800f1de61cd032f89892b183adf8220b1c5cd298"
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/cb97010/ehco/1.1.2-go-1.20-build.patch"
```
Comment (ZhongRuoyu) [inline Formula/e/ehco.rb:32]:
```suggestion
    url "https://raw.githubusercontent.com/Homebrew/formula-patches/cb97010770c6678102b3302e904bb32c6169311c/ehco/1.1.2-go-1.20-build.patch"
```

File: Formula/e/ehco.rb [hunk 2]
```diff
@@ -22,22 +23,21 @@ class Ehco < Formula
...
   end
 
-  depends_on "go@1.19" => :build
+  depends_on "go@1.20" => :build
 
   uses_from_macos "netcat" => :test
 
-  # Backport quic-go update to support Go 1.19
-  # Remove in the next release
+  # go@1.20 build patch, remove in next release
```
Comment (ZhongRuoyu) [inline Formula/e/ehco.rb:30]: If this is already fixed upstream, can we move it to a `stable` block?

File: Formula/e/ehco.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/e/ehco.rb:32]: why do prefer the long hash?

File: Formula/e/ehco.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/e/ehco.rb:30]: yeah, that sounds about right

File: Formula/e/ehco.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/e/ehco.rb]: nb i did not change the hash, as it is the same thing :)

Comment (ZhongRuoyu) [reply Formula/e/ehco.rb]: When the repo has sufficiently long history, the chance of getting clashes in short hashes may no longer be negligible. Just wanted to wipe out any potential ambiguity. But ok, I'm fine with that too.

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**